### PR TITLE
feat(extras): Add scala support using nvim-metals

### PR DESCRIPTION
**Pull request by seakayone**
nvim-metals:
https://github.com/scalameta/nvim-metals

minimal example configuration from their documentation: 
https://github.com/scalameta/nvim-metals/discussions/39

### CODE REVIEW

**Comment by seakayone** [comment]
@artsbentley thank you for sending your love. 

Have you been able to test this out? 
Does it work for you?

**Comment by callmestech** [comment]
I updated metals and it works, sorry 🙈

**Comment by callmestech** [comment]
Every time I search backwards with S, I get this error

**Comment by callmestech** [comment]
When I search backwards using S I have a following error 
```
E5108: Error executing lua: ...0.9.2/share/nvim/runtime/lua/vim/treesitter/language.lua:94: no parser for 'scala' language, see :help treesitter-parsers
stack traceback:
	[C]: in function 'error'
	...0.9.2/share/nvim/runtime/lua/vim/treesitter/language.lua:94: in function 'add'
	...2/share/nvim/runtime/lua/vim/treesitter/languagetree.lua:98: in function 'new'
	...r/neovim/0.9.2/share/nvim/runtime/lua/vim/treesitter.lua:61: in function '_create_parser'
	...r/neovim/0.9.2/share/nvim/runtime/lua/vim/treesitter.lua:131: in function 'get_parser'
	...re/nvim/lazy/flash.nvim/lua/flash/plugins/treesitter.lua:21: in function 'get_nodes'
	...re/nvim/lazy/flash.nvim/lua/flash/plugins/treesitter.lua:87: in function 'fn'
	.../share/nvim/lazy/flash.nvim/lua/flash/search/matcher.lua:39: in function 'get'
	...34/.local/share/nvim/lazy/flash.nvim/lua/flash/cache.lua:87: in function 'get_state'
	...34/.local/share/nvim/lazy/flash.nvim/lua/flash/state.lua:283: in function '_update'
	...34/.local/share/nvim/lazy/flash.nvim/lua/flash/state.lua:248: in function 'update'
	...34/.local/share/nvim/lazy/flash.nvim/lua/flash/state.lua:140: in function 'new'
	...4/.local/share/nvim/lazy/flash.nvim/lua/flash/repeat.lua:51: in function 'get_state'
	...re/nvim/lazy/flash.nvim/lua/flash/plugins/treesitter.lua:97: in function 'treesitter'
	...l/share/nvim/lazy/LazyVim/lua/lazyvim/plugins/editor.lua:245: in function <...l/share/nvim/lazy/LazyVim/lua/lazyvim/plugins/editor.lua:245>
```
Do you know how can I fix it ?

**Comment by amaanq** [comment]
:TSInstall scala probably

**Comment by callmestech** [comment]
The thing is - it's perfectly fine when I search forwards using s

> :TSInstall scala probably

Thank you. I've tried it, but it doesn't work. Interestingly, the forward search using s works fine.

**Comment by amaanq** [comment]
file a bug at flash.nvim would be my next step unless this only happens with "nvim-metals" installed

**Comment by seakayone** [comment]
@callmestech I cannot reproduce you problem with my setup. Searching with `S` works as expected.
Today I have updated all installed plugins with lazy.nvim. Maybe the bug is already solved?

File: lua/lazyvim/plugins/extras/lang/scala.lua [hunk 1]
```diff
@@ -0,0 +1,68 @@
+-- NOTE: This setup does not define any lsp or dap specific key bindings
+-- It is recommended that the LazyVim extras dap-core is imported "lazyvim.plugins.extras.dap.core"
+-- If you like you can setup your own key bindings.
+-- For minimalistic setup have a look at https://github.com/scalameta/nvim-metals/discussions/39
+return {
```
Comment (robsonpeixoto) [inline lua/lazyvim/plugins/extras/lang/scala.lua:5]: Is missing `nvim-treesitter` configuration:
```
  {
    "nvim-treesitter/nvim-treesitter",
    opts = function(_, opts)
      vim.list_extend(opts.ensure_installed, { "scala" })
    end,
  },

```

Comment (seakayone) [reply lua/lazyvim/plugins/extras/lang/scala.lua:5]: Thanks for the hint. 

Until now I have just used the syntax highlighting from Metals.

I agree that using the treesitter highlighting will provide a better experience for most users.

[01f3bbf](https://github.com/LazyVim/LazyVim/pull/1347/commits/01f3bbf9e64e8c1fe2a25cc1d83c02a60bb2d0a9)

**Comment by folke** [comment]
Thanks!

**Comment by seakayone** [comment]
For those interested: I have added a followup PR to improve the experience a bit: https://github.com/LazyVim/LazyVim/pull/3348

**Pull request merged**